### DISCUSSION TITLE
useMatches not changing with URL params like paging and search

### DISCUSSION THREAD

**Discussion by dalalmj** [remix-run/remix#3008] [Q&A] [answered]
Trying to use 'useMatches' to show data for child when user clicks a link on list item. List of items is retried using offset paging, using Link to navigate

```JS
<Link to={`/catalog?skip=${skip}&take=${take}`} >
 Next
</Link>
```
useMatches provides proper data at this list level, but if you try to use it in a child route (in $catalogId.tsx), it always returns data from first page /catalog?skip=0&take=20, no matter whats the current URL /catalog?skip=20&take=20. 

It seems useMatches uses useMemo and it does not have any dependency on url searchParameters. 

Routes
catalog.tsx
/catalog/$catalogid.tsx

```JS
export default function CatalogItem(props: Props) {
  const matches = useMatches();
  const { params, data } = matches.find((m) => m.pathname === "/catalog");
  console.log(`matches: ${JSON.stringify(matches)}`);
```
data always returns first page items, not the current page items...

**Comment by dalalmj** [comment, marked as answer]
Found out the issue. For child item, call to display child is using following Link
```JS
<Link to={`${item.catalogId}`} >
 Details
</Link>
```
And remix ends up reloading "all" routes, including parent route (/catalog) since URL changed, and since it is missing paging searchParams, parent route loads the first page! Solution is to pass searchParams on Link so as to properly reload all routes.

**Reply by kaitlyngreve** [reply]
Hey @dalalmj! I'm struggling with the same issue - can you share your solution code where you use searchParams on Link? Thank you!
